lib/logstash/inputs/rabbitmq/march_hare.rb

```diff
@@ -112,6 +112,9 @@ def consume
       @consumer = @q.build_consumer(:block => true) do |metadata, data|
         @codec.decode(data) do |event|
           decorate(event)
+	  metadata.headers.each do |header, value|
+            event[header] = value.respond_to?('toString') ? value.toString : value 
+         end if @include_headers
```

Comment:
I tested using the rabbitmq management console, it allowed String, Number and List. I manually tested each type and they seem to work... the List "type" ended up getting flattened to a String.